### ISSUE TITLE
Contributing to Abricotine

### ISSUE THREAD

**Issue by Tokelin** [brrd/abricotine#310]
Hello,

I've been using this great editor for a few months now (even to draft this) and I can only say good things about it. Recently I've been interested with doing some open source contributions and since it's an Electron app I feel it's something that should be easy for me to do. However I noticed this comment of yours on one of the PRs - https://github.com/brrd/abricotine/pull/293#issuecomment-725407871. I would love to contribute to this project, but - ultimately - I just want to be reassured that I won't be left to endlessly open PRs with no chance of merging. Currently I'm in the process of getting the code running locally (Windows and node-gyp are... less than cooperative to put it nicely 😁) with the intent to poke around and see where I can improve, enhance or fix something. Of course I am open to feedback in the sense of issue priority and such.

Thank you for your time and this amazing project.

**Comment by brrd** [comment]
**TL;DR: Abricotine is going to be discontinued. Please don't waste your time fixing it.**

Thanks you for your interest in Abricotine.

As you noticed, I'm today involved in other projects and I lack time and motivation to keep on maintaining this application. Moreover I must confess that I personally switched to another markdown editor, which makes the project even less appealing to me.

For months (years?) I'm doing my best to provide a minimum support and maintenance because I know Abricotine still have some enthusiastic users and contributors, and feel kind of guilty to give them up :sweat:

(By the way, I thank them very much for their support :pray:)

But to be honest **I'm thinking of dropping this project soon**. And even if I could find the motivation to keep on working on it, I would probably rewrite most of the code (which appears to be 50% garbage from my today point of view). Anyway, I definitely don't want to spend my time reviewing changes in the current code. I'm sorry.

Of course you can decide to create and maintain your own fork. But before you spend more of your time on it, you need to know that most of the app was written six years ago, when I was a total JavaScript beginner. At this time it was supposed to be nothing more than a training project. I could not find any markdown editor featuring inline preview, so I decided to create mine and publish it as an open-source software.

Things changed a lot since then. I personally learned how to be a better developer (or at least I hope so). The JavaScript ecosystem moved to frameworks, Typescript and ES6. Electron and NodeJs improved a lot. Github released Github Actions. [Marijn Haverbeke](https://github.com/marijnh) (the creator of CodeMirror) developed the [ProseMirror](https://prosemirror.net/) framework and some [very interesting derived editors](https://milkdown.dev/) appeared. Etc.

And the most important, plenty of very good alternatives to Abricotine came to the market:

* [Zettlr](https://www.zettlr.com/)
* [Marktext](https://marktext.app/)
* [Ghostwriter](http://wereturtle.github.io/ghostwriter/)
* [Typora](https://typora.io) (closed source) 
* [And many more...](https://alternativeto.net/software/abricotine/)

So my advice would be either to contribute to one of those projects instead (at least two of them use Electron) or to write your own editor if you don't like them. But I don't really like the idea that developers will waste their precious time fixing my old dirty code...

I'm sorry for this answer which is probably not the one your wanted to read.

By the way, I'm taking the freedom to pin this issue to increase its visibility.

**Comment by Tokelin** [comment]
Thank you for your openness and the recommendations for other Markdown editors. I cannot lie and say I'm not disappointed by the response, but I can completely understand it.

I wish you all the best in your future endeavours!

**Comment by KorbsStudio** [comment]
Hmm...I would love to continue this project. Anything we can do?

**Comment by brownc0** [comment]
I mainly use Abricotine because of it's "copy as HTML" feature. Do know any markdown editors that have this feature?

**Comment by KorbsStudio** [comment]
I also wanted something like that for website building, but Jekyll exist

**Comment by granbosque** [comment]
I don't like you calling it old dirty code... 😅
I have not found any other editor that reunites all of these features:

- i18n (not in marktext)
- inline display of images (not in ghostwriter)
- Themeable and editing directly in markdown, without need to switch between edit and preview mode.
- Good interface 

It lacks some features, but I think it's one of the most usable editor... also thinking in contributing or forking.

**Comment by brrd** [comment]
Abricotine is discontinued. See #347.
